eoc_journal/eoc_journal.py

```diff
@@ -97,6 +106,49 @@ def student_view(self, context=None):
         )
         return fragment
 
+    def list_user_pb_answers_by_section(self):
+        """
+        Returns a list of dicts with pb-answers grouped by section.
+        """
+        # Get the selected blocks and their answers
+        blocks = self.list_pb_answers()
+        course_id = self._get_course_id()
+        user_id = self.runtime.anonymous_student_id
+        answers_names = [block['name'] for block in blocks]
+
+        answers = Answer.objects.filter(  # pylint: disable=no-member
+            course_key=course_id,
+            student_id=user_id,
+            name__in=answers_names
+        )
+
+        if answers.count() > 0:
+            # Map answer names to student inputs
+            students_inputs = {a.name: a.student_input for a in answers if a.student_input}
+
+            # Group answers by section
+            answers = OrderedDict()
+
+            for block in blocks:
+                name = block['name']
+                section = block['section']
+
+                if section not in answers:
+                    answers[section] = []
+
+                answers[section].append({
+                    'answer': students_inputs[name],
```

Comment:
It was meant to indicate that no answer was given while still displaying the question.

However there are probably better ways to do that and since there is currently not a clear requirement on what to do when question hasn't been answer yet, I think it's fine to just omit the question/answer from the list in that case for now.